### PR TITLE
initial commit for cmek rotation

### PR DESCRIPTION
**Pull request by kevensen**
Initial commit.  Added ability to parameterize the rotation duration.

### CODE REVIEW

File: policies/templates/gcp_cmek_rotation_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+           deny[{
+           	"msg": message,
+           	"details": metadata,
+           }] {
+           	constraint := input.constraint
+           	lib.get_constraint_params(constraint, params)
+           	asset := input.asset
+           	asset.asset_type == "cloudkms.googleapis.com/CryptoKey"
+           
+           	rotation_period_string := lib.get_default(asset.resource.data, "rotationPeriod", "99999999s")
```
Comment (morgante) [inline policies/templates/gcp_cmek_rotation_v1.yaml]: Could we provide an explanation/comment for the default here?

File: policies/templates/gcp_cmek_rotation_v1.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+           	rotation_period := time.parse_duration_ns(rotation_period_string)
+           
+           	period_string := lib.get_default(params, "period", "31536000s")
+           	period_to_test := time.parse_duration_ns(period_string)
+           
+           	rotation_period > period_to_test
+           	message := sprintf("%v: CMEK Rotation Period must be or less.", [asset.name])
```
Comment (morgante) [inline policies/templates/gcp_cmek_rotation_v1.yaml]: Violation phrasing seems off, maybe "CMEK rotation period must be at most %v"

File: samples/cmek_rotation.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+metadata:
+  name: cmek_rotation
+spec:
+  severity: high
+  match:
+    gcp:
+      target: ["organization/*"]
+  parameters:
+    # Optionaly specify the required key rotation period.  Default is 365 days
+    # Valud time units are  "ns", "us" (or "µs"), "ms", "s", "m", "h"
```
Comment (morgante) [inline samples/cmek_rotation.yaml]:
```suggestion
    # Valid time units are  "ns", "us" (or "µs"), "ms", "s", "m", "h"
```

File: policies/templates/gcp_cmek_rotation_v1.yaml (hunk 1 above)
Comment (kevensen) [reply policies/templates/gcp_cmek_rotation_v1.yaml]: done

File: policies/templates/gcp_cmek_rotation_v1.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+           	# from the CAI.  The default is 99999999s.  This is 
+           	# sufficiently high enough to cause fail
+           	rotation_period_string := lib.get_default(asset.resource.data, "rotationPeriod", "99999999s")
+           	rotation_period := time.parse_duration_ns(rotation_period_string)
+           
+           	period_string := lib.get_default(params, "period", "31536000s")
+           	period_to_test := time.parse_duration_ns(period_string)
+           
+           	rotation_period > period_to_test
+           	message := sprintf("%v: CMEK Rotation Period must be at least %v.", [asset.name, period_string])
```
Comment (morgante) [inline policies/templates/gcp_cmek_rotation_v1.yaml]: Isn't it the opposite?

```suggestion
           	message := sprintf("%v: CMEK Rotation Period must be at most %v.", [asset.name, period_string])
```

Comment (kevensen) [reply policies/templates/gcp_cmek_rotation_v1.yaml]: yup!

**Pull request merged**